compiler/src/main/java/io/neow3j/compiler/NeoMethod.java

```diff
@@ -465,13 +465,37 @@ public NeoVariable getParameterByJVMIndex(int index) {
      */
     public void convert(CompilationUnit compUnit) throws IOException {
         AbstractInsnNode insn = asmMethod.instructions.get(0);
+        throwIfMethodHasSdkRelatedReturnOrArgType(asmMethod);
         while (insn != null) {
             insn = Compiler.handleInsn(insn, this, compUnit);
             insn = insn.getNext();
         }
         insertTryCatchBlocks();
     }
 
+    private void throwIfMethodHasSdkRelatedReturnOrArgType(MethodNode asmMethod) {
+        Type returnType = Type.getReturnType(asmMethod.desc);
+        if (isNonDevpackNeow3jType(returnType.getClassName())) {
+            throw new CompilerException(
+                    format("The neow3j compiler does not support SDK-related types. Type '%s' used as return type of " +
+                            "method '%s' is not supported.", returnType.getInternalName(), asmMethod.name));
+        }
+        for (Type argType : Type.getArgumentTypes(asmMethod.desc)) {
```

Comment:
Does this also cover method-local variables like the following?
`io.neow3j.types.Hash160 localHashVar = new io.neow3j.types.Hash160(...);`

Reply:
Do we have tests for this? If not, we need it.

Reply:
@mialbu what's the state of this? what @csmuller suggested/asked is possible to be done? or is already covered?